src/lustre/lustreRemoveMultAssign.ml

```diff
@@ -0,0 +1,299 @@
+        let* res1 = R.seq (List.map (helper ctx) l1) in
+        let nis1, nis2, decls1, ctx1 = split_and_flatten4 res1 in
+        let* res2 = R.seq (List.map (helper ctx) l2) in
+        let nis3, nis4, decls2, ctx2 = split_and_flatten4 res2 in
+        (* nis1 and nis3 are the temp variables need to get pulled outside the if block *)
+        R.ok (nis1 @ nis3, [A.IfBlock (pos, e, nis2, nis4)], decls1 @ decls2, ctx1 @ ctx2)
+
+      | FrameBlock (pos, vars, nes, nis) -> 
+        let nes = List.map (fun x -> A.Body x) nes in 
+        let* res1 = R.seq (List.map (helper ctx) nes) in
+        let nis1, nis2, decls1, ctx1 = split_and_flatten4 res1 in
+        let* res2 = R.seq (List.map (helper ctx) nis) in
+        let nis3, nis4, decls2, ctx2 = split_and_flatten4 res2 in     
+        let nis2 = List.map 
+          (fun x -> match x with | A.Body (Equation _ as e) -> e | _ -> assert false) 
+          nis2 in
+        (* nis1 and nis3 are the temp variables need to get pulled outside the if block *)
+        R.ok (nis1 @ nis3, [A.FrameBlock (pos, vars, nis2, nis4)], decls1 @ decls2, ctx1 @ ctx2)
+      
+      (* Misplaced assert, annotmain, annotproperty in if block*)
+      | A.Body (Assert (pos, _)) 
+      | A.AnnotProperty (pos, _, _)
+      | A.AnnotMain (pos, _) -> mk_error pos (MisplacedNodeItemError ni)
+  ) in
+  let* (nis, nis2, new_decls, ctx) = helper ctx ni in
+  let ctx = List.fold_left Ctx.union Ctx.empty_tc_context ctx in
+  (* Calling 'remove_mult_assign_from_ib' on an if or frame block (which is always
+     the case) will mean that nis2 will always have length 1. *)
+  let ni = List.hd nis2 in
+  R.ok (nis, ni, new_decls, ctx)
+
+let desugar_node_item ctx ni = match ni with
+  | A.IfBlock _ -> 
+    let* (nis, ib, decls1, _) = remove_mult_assign_from_ni ctx ni in 
+    R.ok (decls1, nis @ [ib])
+  | A.FrameBlock _ -> 
+    let* (nis2, fb, decls1, _) = remove_mult_assign_from_ni ctx ni in 
+    R.ok (decls1, nis2 @ [fb])
+  | _ -> R.ok ([], [ni])
+
+(** Desugars an individual node declaration (removing IfBlocks). *)
+let desugar_node_decl ctx decl = match decl with
+  | A.FuncDecl (s, ((node_id, b, nps, cctds, ctds, nlds, nis, co) as d)) ->
+    let* ctx = Chk.get_node_ctx ctx d in
+    let* res = R.seq (List.map (desugar_node_item ctx) nis) in
+    let (new_decls, nis) = List.split res in
+    R.ok (A.FuncDecl (s, (node_id, b, nps, cctds, ctds, (List.flatten new_decls) @ nlds, (List.flatten nis), co))) 
+  | A.NodeDecl (s, ((node_id, b, nps, cctds, ctds, nlds, nis, co) as d)) -> 
+    let* ctx = Chk.get_node_ctx ctx d in
+    let* res = R.seq (List.map (desugar_node_item ctx) nis) in
+    let (new_decls, nis) = List.split res in
+    R.ok (A.NodeDecl (s, (node_id, b, nps, cctds, ctds, (List.flatten new_decls) @ nlds, (List.flatten nis), co))) 
+  | _ -> R.ok (decl)
+
+
+(** Desugars a declaration list to remove IfBlocks. Converts IfBlocks to
+    declarative ITEs, filling in oracles if branches are undefined. *)
```

Comment:
This comment does not belong here. Update comment to reflect the behavior of this function.